src/manifestoo/git.py

```diff
@@ -0,0 +1,65 @@
+import re
+import subprocess
+from pathlib import Path
+from typing import Callable, List
+
+
+class BranchNotFound(Exception):
+    pass
+
+
+def get_current_branch() -> str:
+    command_current = ["git", "rev-parse", "--abbrev-ref", "HEAD"]
+    return subprocess.check_output(command_current).strip().decode()
+
+
+def check_branch_exists(branch: str) -> None:
+    try:
+        command_check = ["git", "rev-parse", "--verify", branch]
+        subprocess.check_output(command_check)
+    except subprocess.CalledProcessError:
+        raise BranchNotFound(f"Cannot find branch {branch}. Aborting.")
+
+
+def _get_new_manifest_files(branch: str, current_branch: str) -> List[str]:
+    command_new = [
+        "git",
+        "diff",
+        "--name-only",
+        "--diff-filter",
+        "A",
+        branch,
+        current_branch,
+    ]
+    new_files = subprocess.check_output(command_new).decode().split()
+    return [f for f in new_files if f.endswith("/__manifest__.py")]
+
+
+def get_branch_new_addons(branch: str, addons_dirs: List[Path]) -> List[str]:
+    return _get_branch_addons_by(_get_new_manifest_files, branch, addons_dirs)
+
+
+def _get_modified_files(branch: str, current_branch: str) -> List[str]:
+    command_modified = ["git", "diff", "--name-only", branch, current_branch]
+    return subprocess.check_output(command_modified).decode().split()
+
+
+def get_branch_modified_addons(branch: str, addons_dirs: List[Path]) -> List[str]:
+    return _get_branch_addons_by(_get_modified_files, branch, addons_dirs)
+
+
+def _get_branch_addons_by(
+    method: Callable[[str, str], list[str]], branch: str, addons_dirs: List[Path]
```

Comment:
I'm not sure how to get tox to be happy with the typing of Callable. It seems any syntax fails in some version of python.
Interestingly enough, passing Callable without any details work in local against all python, but apparently not here. 
Passing type: ignore or Any because of that is quite unpalatable, to say the least, so I'm not sure how to handle that.

Reply:
I've made the change anyway because I can't even work with this, since the linting rewrite the code every time I work on it. This feels like a rube-Goldberg machine where every part incorrects one another.